cpp/src/groupby/hash/compute_mapping_indices.cuh

```diff
@@ -42,6 +42,11 @@ __device__ void find_local_mapping(cooperative_groups::thread_block const& block
         auto const ref_cardinality =
           cuda::atomic_ref<size_type, cuda::thread_scope_block>{*cardinality};
         auto const shared_set_index = ref_cardinality.fetch_add(1, cuda::std::memory_order_relaxed);
+
+        // The value of shared_set_index is before increment, thus we will have the
+        // cardinality exceeded its threshold value.
```

Comment:
I'm not sure if I completely understand this. Can we reword? The value is `<something>` before it is incremented, and the cardinality will exceed the threshold.